mgmt/rest/server.go

```diff
@@ -299,8 +309,20 @@ func (s *Server) Name() string {
 	return "REST"
 }
 
-func (s *Server) SetAddress(addrString string) {
-	s.addrString = addrString
+func (s *Server) SetAddress(addrString string, dfltPort int) error {
+	restLogger.Info(fmt.Sprintf("Setting address to: [%v] Default port: %v", addrString, dfltPort))
+	// In the future, we could extend this to support multiple comma separated IP[:port] values
+	if strings.Index(addrString, ",") != -1 {
+		return serror.ErrBadAddress
```

Comment:
Understood and agreed. However I will still run the test and call log.Fatal because we might enter this code when running tests like PluginRestCalls for instance which will not go through

Considering this, I would therefore like to "factorize/refactor/modify" the checkCmdLineFlags and extract from it the address validation mechanisms so that this can be common to both snapd.go and server.go. However, where should I put such a code in the current code hierarchy ?